shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1815,6 +1815,79 @@ void WebContents::OnFirstNonEmptyLayout(
   }
 }
 
+// This object wraps the InvokeCallback so that if it gets GC'd by V8, we can
+// still call the callback and send an error. Not doing so causes a Mojo DCHECK,
+// since Mojo requires callbacks to be called before they are destroyed.
+class ReplySender : public gin::Wrappable<ReplySender> {
+ public:
+  using InvokeCallback = electron::mojom::ElectronApiIPC::InvokeCallback;
+  static gin::Handle<ReplySender> Create(v8::Isolate* isolate,
+                                         InvokeCallback callback) {
+    return gin::CreateHandle(isolate, new ReplySender(std::move(callback)));
+  }
+
+  // gin::Wrappable
+  static gin::WrapperInfo kWrapperInfo;
+  gin::ObjectTemplateBuilder GetObjectTemplateBuilder(
+      v8::Isolate* isolate) override {
+    return gin::Wrappable<ReplySender>::GetObjectTemplateBuilder(isolate)
+        .SetMethod("sendReply", &ReplySender::SendReply);
+  }
+  const char* GetTypeName() override { return "ReplySender"; }
+
+ private:
+  explicit ReplySender(InvokeCallback callback)
+      : callback_(std::move(callback)) {}
+  ~ReplySender() override {
+    if (callback_) {
+      v8::Isolate* isolate = electron::JavascriptEnvironment::GetIsolate();
+      // If there's no current context, it means we're shutting down, so we
+      // don't need to send an event.
+      if (!isolate->GetCurrentContext().IsEmpty()) {
+        v8::HandleScope scope(isolate);
+        auto message = gin::DataObjectBuilder(isolate)
+                           .Set("error", "reply was never sent")
+                           .Build();
+        SendReply(isolate, message);
```

Comment:
Is this new behavior? Sending a default reply when the sender is GC'ed?

Reply:
nope, this used to be implemented on `Event`. I moved it here because it's only needed for the IPC events, which contain a reference to a Mojo reply callback, which must be invoked before they are destroyed.